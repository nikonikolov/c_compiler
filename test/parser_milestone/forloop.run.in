int glob1=10;
int glob2=67;
int glob3=80;
int glob4=0;
int fn(int a, int b, int c, int d, int e, int f, int g){
 	int sum = 0;
 	for(a=0;a<glob2;a++){
 		sum+=a;
 	}
 	return sum;
}
