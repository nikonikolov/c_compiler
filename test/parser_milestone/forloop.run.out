VARIABLE : glob1
VARIABLE : glob2
VARIABLE : glob3
VARIABLE : glob4
FUNCTION : fn
    PARAM : a
    PARAM : b
    PARAM : c
    PARAM : d
    PARAM : e
    PARAM : f
    PARAM : g
SCOPE
    VARIABLE : sum
    FOR LOOP
    INIT BLOCK: (a)=(0)
    CONDITION BLOCK: (a)<(glob2)
    UPDATE BLOCK: (a)++
    SCOPE
        (sum)+=(a)
    RETURN sum
